pkg/metadata/provider_packet.go

```diff
@@ -84,6 +88,28 @@ func (p *ProviderPacket) Extract() ([]byte, error) {
 
 // networkConfig handles Packet network configuration, primarily bonding
 func networkConfig(ni metadata.NetworkInfo) error {
+	// set up bonding
+	bond, err := netlink.LinkByName("bond0")
+	if err != nil {
+		if strings.Contains(err.Error(), "Link not found") {
+			newBond := new(netlink.Bond)
+			newBond.LinkAttrs.Name = "bond0"
+			newBond.Mode = netlink.BondMode(ni.BondingMode())
+
+			if err := netlink.LinkAdd(newBond); err != nil {
+				if !strings.Contains(err.Error(), "file exists") {
+					return fmt.Errorf("Could not make new bonded device: %s", err)
+				}
+			}
+			bond, err = netlink.LinkByName("bond0")
+			if err != nil {
+				return fmt.Errorf("Could not get link by name for bond0: %s", err)
+			}
+		} else {
+			return fmt.Errorf("Could not get link by name for bond0: %s", err)
+		}
+	}
```

Comment:
it would make sense to print a message saying "using existing bond0 interface" in this case.